pkg/v1/tkg/managementcomponents/management_component_install.go

```diff
@@ -48,18 +60,209 @@ type ManagementComponentsInstallOptions struct {
 	ManagementPackageRepositoryOptions ManagementPackageRepositoryOptions
 }
 
+func generateAddonSecretName(clusterName, addonName string) string {
+	return fmt.Sprintf("%s-%s-addon", clusterName, fmt.Sprintf("tanzu-%s", addonName))
+}
+
+func pauseAddonSecretReconciliation(ctx context.Context, clusterClient clusterclient.Client, addonSecreteName, namespace string) error {
+	log.Infof("Pausing reconciliation for %s/%s secret", namespace, addonSecreteName)
+	secret := &corev1.Secret{}
+	jsonPatch := []map[string]interface{}{
+		{
+			"op":    "add",
+			"path":  fmt.Sprintf("/metadata/annotations/tkg.tanzu.vmware.com~1addon-paused"),
+			"value": "",
+		},
+	}
+	payloadBytes, err := json.Marshal(jsonPatch)
+	if err != nil {
+		return errors.Wrap(err, "unable to generate json patch")
+	}
+
+	err = clusterClient.PatchResource(secret, addonSecreteName, namespace, string(payloadBytes), types.JSONPatchType, nil)
+	if apierrors.IsNotFound(err) {
```

Comment:
Question: Are we ignoring not found errors everywhere because we want the cluster upgrade to be repeatable in case of errors?

Reply:
yes. Idempotency. If we hit a "not found" it means the resource does not exists so we have to do nothing. 

The last thing we do is delete the secret which will make a new execution skip all of this.